cocos/spine/cocos-spine/spine-skeleton-animation.ts

```diff
@@ -0,0 +1,488 @@
+    get useTint () {
+        return this._useTint;
+    }
+    set useTint (val) {
+        if (this._useTint !== val) {
+            this._useTint = val;
+        }
+        this._updateUseTint();
+    }
+
+    /**
+     * @en Whether play animations in loop mode.
+     * @zh 是否循环播放当前骨骼动画。
+     */
+    @type(CCBoolean)
+    @tooltip('i18n:COMPONENT.skeleton.loop')
+    get loop () {
+        return this._loop;
+    }
+    set loop (val) {
+        this._loop = val;
+    }
+
+    /**
+     * @en Whether premultipliedAlpha enabled.
+     * @zh 是否启用 alpha 预乘。
+     */
+    @editable
+    @tooltip('i18n:COMPONENT.skeleton.premultipliedAlpha')
+    get premultipliedAlpha (): boolean { return this._premultipliedAlpha; }
+    set premultipliedAlpha (v: boolean) {
+        if (v !== this._premultipliedAlpha) {
+            this._premultipliedAlpha = v;
+            this._skeleton.setPremultipliedAlpha(this._premultipliedAlpha);
+            //if (this._renderer) this._renderer.premultipliedAlpha = v;
+        }
+    }
+
+    @editable
+    @range([0, 10.0])
+    @type(CCFloat)
+    @tooltip('i18n:Animation Speed')
+    @displayName('Time Scale')
+    get timeScale (): number {
+        return this._timeScale;
+    }
+    set timeScale (val: number) {
+        this._timeScale = val;
+        this.updateTimeScale(val);
+    }
+
+    @type([SpineSocket])
+    @tooltip('i18n:SpineBone.sockets')
+    get sockets (): SpineSocket[] {
+        return this._sockets;
+    }
+    set sockets (val: SpineSocket[]) {
+        this._sockets = val;
+        this._updateSocketBindings();
+        this._syncAttachedNode();
+    }
+
+    get skinName () {
+        return this._skinName;
+    }
+    public setSkin (skinName: string) {
+        if (this._skinName === skinName) return;
+
+        this._skinName = skinName;
+        this._skeleton.setSkin(skinName);
+        this._updateRenderData();
+    }
+
+    // update skin list for editor
+    protected _updateSkinEnum () {
+        if (!EDITOR) return;
+        let skinEnum;
+        if (this.skeletonData) {
+            skinEnum = this.skeletonData.getSkinsEnum();
+        } else {
+            skinEnum = SpineSkinEnum;
+        }
+
+        const enumSkins = Enum({});
+        Object.assign(enumSkins, skinEnum);
+        Enum.update(enumSkins);
+        setEnumAttr(this, 'defaultSkinIndex', enumSkins);
+    }
+
+    // update animation list for editor
+    protected _updateAnimEnum () {
+        let animEnum;
+        if (this.skeletonData) {
+            animEnum = this.skeletonData.getAnimsEnum();
+        } else {
+            animEnum = SpineAnimationEnum;
+        }
+        // reset enum type
+        const enumAnimations = Enum({});
+        Object.assign(enumAnimations, animEnum);
+        Enum.update(enumAnimations);
+        setEnumAttr(this, 'animationIndex', enumAnimations);
+    }
+
+    public __preload () {
+        if (EDITOR) {
+            this._updateSkinEnum();
```

Comment:
I don't think this is a good implementation, try to implement it through custom inspector